src/promptflow/tests/sdk_cli_test/e2etests/test_cli_time_consume.py

```diff
@@ -0,0 +1,115 @@
+import contextlib
+import io
+import sys
+import tempfile
+import uuid
+from pathlib import Path
+import pytest
+import cProfile
+import pstats
+from promptflow._cli._pf.entry import main
+import multiprocessing
+
+FLOWS_DIR = "./tests/test_configs/flows"
+DATAS_DIR = "./tests/test_configs/datas"
+
+
+def run_cli_command(cmd, time_limit=3600, is_print_stats=True, *, result_queue=None):
+    sys.argv = list(cmd)
+    with cProfile.Profile() as pr:
+        output = io.StringIO()
+        with contextlib.redirect_stdout(output):
+            main()
+
+        pstats_obj = pstats.Stats(pr).sort_stats(pstats.SortKey.CUMULATIVE)
+        stats_profile = pstats_obj.get_stats_profile()
+        if is_print_stats:
+            print(pstats_obj.print_stats(50))
+        assert stats_profile.total_tt < time_limit
+
+        res_value = output.getvalue()
+        if result_queue:
+            result_queue.put(res_value)
+        return res_value
+
+
+def subprocess_run_cli_command(cmd, time_limit=3600, is_print_stats=True):
+    result_queue = multiprocessing.Queue()
+    process = multiprocessing.Process(target=run_cli_command, args=(cmd,), kwargs={"time_limit": time_limit,
+                                                                                   "is_print_stats": is_print_stats,
+                                                                                   "result_queue": result_queue})
+    process.start()
+    process.join()
+    assert process.exitcode == 0
+    return result_queue.get_nowait()
+
+
+@pytest.mark.usefixtures("use_secrets_config_file", "setup_local_connection")
+@pytest.mark.cli_test
+@pytest.mark.e2etest
+class TestCliTimeConsume:
```

Comment:
Let's add test for connection list/get, which are the most calls triggered by VSC Extension

Reply:
done